### PR TITLE
Add superuser keyboard definitions.

### PR DESCRIPTION
**Pull request by kaylanm**
## Description

Add superuser tkl, ftl, ext keyboard definitions & default + via keymaps.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* N/A

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/superuser/ext/config.h [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x5355
+#define PRODUCT_ID      0x4558
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    superuser
+#define PRODUCT         ext
+
+#define CONFIG_H
+
+#include "config_common.h"
```
Comment (drashna) [inline keyboards/superuser/ext/config.h:27]:
```suggestion
```

File: keyboards/superuser/ext/config.h [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+/* USB Device descriptor parameter */
```
Comment (drashna) [inline keyboards/superuser/ext/config.h:18]:
```suggestion
#pragma once

#include "config_common.h"

/* USB Device descriptor parameter */
```

File: keyboards/superuser/ext/ext.c [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/superuser/ext/ext.c:16]:
```suggestion
#include "ext.h"
```

File: keyboards/superuser/ext/readme.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+superuser ext
+=============
```
Comment (drashna) [inline keyboards/superuser/ext/readme.md:2]:
```suggestion
# superuser ext
```

File: keyboards/superuser/ext/readme.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+An EXT TKL PCB supporting 7u bottom row only.
+
+Keyboard Maintainer: superuser
+Hardware Supported: superuser ext
+Hardware Availability: Group buy
```
Comment (drashna) [inline keyboards/superuser/ext/readme.md:8]:
```suggestion
* Keyboard Maintainer: superuser
* Hardware Supported: superuser ext
* Hardware Availability: Group buy
```

File: keyboards/superuser/ext/rules.mk [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (drashna) [inline keyboards/superuser/ext/rules.mk:11]:
```suggestion
```

File: keyboards/superuser/ext/rules.mk [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = no        # Enable RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+EXTRAFLAGS += -flto
```
Comment (drashna) [inline keyboards/superuser/ext/rules.mk:32]:
```suggestion
LTO_ENABLE = yes
```

File: keyboards/superuser/frl/config.h [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x5355
+#define PRODUCT_ID      0x4652
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    superuser
+#define PRODUCT         frl
+
+#define CONFIG_H
+
+#include "config_common.h"
```
Comment (drashna) [inline keyboards/superuser/frl/config.h:27]:
```suggestion
```

File: keyboards/superuser/frl/config.h [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+/* USB Device descriptor parameter */
```
Comment (drashna) [inline keyboards/superuser/frl/config.h:18]:
```suggestion
#pragma once

#include "config_common.h"

/* USB Device descriptor parameter */
```

File: keyboards/superuser/frl/frl.c [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/superuser/frl/frl.c:16]:
```suggestion
#include "fri.h"
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by kaylanm** [comment]
Thank you for reviewing @drashna! I've made your updates across the board.

**Review by drashna — APPROVED** [review]

**Comment by kaylanm** [comment]
@drashna do I need to do anything else on this PR?

File: keyboards/superuser/tkl/info.json [hunk 11]
```diff
@@ -0,0 +1,12 @@
+{
+    "keyboard_name": "tkl",
+    "url": "",
+    "maintainer": "superuser",
+    "width": 18.25,
+    "height": 6.25,
```
Comment (fauxpark) [inline keyboards/superuser/tkl/info.json:6]:
```suggestion
```

File: keyboards/superuser/frl/info.json [hunk 12]
```diff
@@ -0,0 +1,12 @@
+{
+    "keyboard_name": "frl",
+    "url": "",
+    "maintainer": "superuser",
+    "width": 18.25,
+    "height": 5,
```
Comment (fauxpark) [inline keyboards/superuser/frl/info.json:6]:
```suggestion
```

File: keyboards/superuser/ext/info.json [hunk 13]
```diff
@@ -0,0 +1,12 @@
+{
+    "keyboard_name": "ext",
+    "url": "",
+    "maintainer": "superuser",
+    "width": 20.5,
+    "height": 5,
```
Comment (fauxpark) [inline keyboards/superuser/ext/info.json:6]:
```suggestion
```

File: keyboards/superuser/tkl/rules.mk [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = no        # Enable RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+LTO_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/superuser/tkl/rules.mk:25]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
LTO_ENABLE = yes
```

File: keyboards/superuser/frl/rules.mk [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = no        # Enable RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+LTO_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/superuser/frl/rules.mk:25]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
LTO_ENABLE = yes
```

File: keyboards/superuser/ext/rules.mk [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = no        # Enable RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+LTO_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/superuser/ext/rules.mk:25]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
LTO_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**